### PR TITLE
govc: add cluster.mv command

### LINKED ISSUE #2524: [GOVC] Add a new command to support move host from datacenter to cluster
**Is your feature request related to a problem? Please describe.**
I'd like to use to govc to move a host from datacenter to cluster, have tried to use object.mv option but failed as cluster is not a folder

**Describe the solution you'd like**
Could you help add a new command option to wrapper around MoveInto function in here https://github.com/vmware/govmomi/blob/master/object/cluster_compute_resource.go#L72

The command can help move a host from datacenter to cluster

### PR DESCRIPTION
**Pull request by dougm**
## Description

Move hosts into a cluster.

vcsim: add ClusterComputeResource.MoveInto_Task support

Closes #2524

## Type of change

Please mark options that are relevant:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
  not work as expected)
- [ ] This change requires a documentation update
- [ ] Build related change

## How Has This Been Tested?

added new bats tests

## Checklist:

- [x] My code follows the `CONTRIBUTION`
  [guidelines](https://github.com/vmware/govmomi/blob/main/CONTRIBUTING.md) of
  this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged

### CODE REVIEW

File: govc/test/cluster.bats [hunk 1]
```diff
@@ -327,3 +327,39 @@ _EOF_
...
+  assert_success
+  [ ${#lines[@]} -eq 4 ]
+
+  run govc cluster.mv -cluster DC0_C1 DC0_C0_H*
+  assert_failure
+
+  run govc host.maintenance.enter DC0_C0_H*
+  assert_success
+
+  run govc cluster.mv -cluster DC0_C1 DC0_C0_H*
```
Comment (hickeng) [inline govc/test/cluster.bats:344]: Try moving a single host first and check the cluster split is 5:3

File: simulator/cluster_compute_resource.go [hunk 2]
```diff
@@ -350,6 +350,40 @@ func (c *ClusterComputeResource) ReconfigureComputeResourceTask(ctx *Context, re
...
+					return nil, new(types.InvalidState)
+				}
+
+				RemoveReference(&parent.Host, ref)
+			case *mo.ComputeResource:
+				ctx.Map.Remove(ctx, parent.Self)
+			}
+
+			c.Host = append(c.Host, ref)
+			host.Parent = &c.Self
```
Comment (hickeng) [inline simulator/cluster_compute_resource.go:374]: Is there an event of any kind that should trigger from the host movement?  I don't have a VC to hand - possibly:
* HostRemovedEvent
* HostNotInClusterEvent

File: govc/test/cluster.bats (hunk 1 above)
[the remark this answers is not in the document]
Comment (dougm) [reply govc/test/cluster.bats]: Done

File: govc/test/cluster.bats [hunk 3]
```diff
@@ -327,3 +327,39 @@ _EOF_
...
+  run govc find -type h /DC0/host/DC0_C0
+  assert_success
+  [ ${#lines[@]} -eq 0 ]
+
+  run govc find -type h /DC0/host/DC0_C1
+  assert_success
+  [ ${#lines[@]} -eq 8 ]
+
+  # standalone host
+  run govc cluster.mv -cluster DC0_C1 DC0_H0
```
[the remark this answers is not in the document]
Comment (dougm) [reply govc/test/cluster.bats:374]: cluster.mv (MoveInto_Task) can only move hosts into a cluster (from standalone or another cluster).
object.mv (MoveIntoFolder_Task) can move from a cluster into a host folder and it becomes a standalone host (just learned that tonight). Let's revisit that case, at the moment vcsim only supports moving from Folders:

https://github.com/vmware/govmomi/blob/c71501642a0542effc59f81d97e02543079b203b/simulator/folder.go#L567-L571

File: simulator/cluster_compute_resource.go (hunk 2 above)
Comment (dougm) [reply simulator/cluster_compute_resource.go:374]: I didn't see those or other events triggered by MoveInto_Task using a real VC.

**Review by hickeng — APPROVED** [review]

**Pull request merged**